lib/Default/AbstractSmrPlayer.class.php

```diff
@@ -2222,11 +2238,17 @@ public function &killPlayerByPlayer(AbstractSmrPlayer $killer) {
 			$named_ship = strip_tags($this->getCustomShipName(), '<font><span><img>');
 			$msg .= ' flying <span class="yellow">' . $named_ship . '</span>';
 		}
-		$msg .= ' was destroyed by ' . $killer->getBBLink();
-		if ($killer->hasCustomShipName()) {
-			$named_ship = strip_tags($killer->getCustomShipName(), '<font><span><img>');
-			$msg .= ' flying <span class="yellow">' . $named_ship . '</span>';
+		//adjust message for newbie kill
+		if ($this->isNoncombatNewbie() && !$killer->hasNewbieStatus()) {
+			$msg .= ' was newbie killed by ' . $killer->getBBLink();
+		} else {
+			$msg .= ' was destroyed by ' . $killer->getBBLink();
+			if ($killer->hasCustomShipName()) {
```

Comment:
Ship name logic doesn't need to be (and probably shouldn't be) inside the `else` clause.